### ISSUE TITLE
need opposite of no-delay-provisioning: only start if queue is larger than x builds

### ISSUE THREAD

**Issue by cloph** [jenkinsci/ec2-fleet-plugin#151]
Currently there's no way to configure when the plugin should start increasing the capacity of the fleet.
There should be a way to configure a minimal queue length before starting instances.
Rationale:
First build by a EC2 instance can have significant overhead (like for example no up-to-date ccache that needs to be filled or the initial slower performance when still copying data from S3 snapshot to the EBS volume or similar one-time-per-instance setup operations.
Also queue length of the EC2 workers might not matter in the overall picture so much if there are other jobs to do alongside that allow for some amount of queue growth before final results are bottlenecked by the sepcific job that is supported by a EC2 fleet setup.

Thus I'd like to ensure that the queue is large enough for the EC2-instance to do at least three builds in a row to even out the initial cost over the lifetime of the instance.

The default launching algorithm is too eager to keep the queue low - and thus starts Instances too early and terminates them again after a single build. 

We for example have a multijob configuration with jobs for Windows, Linux and Mac that each have a different number of permanent 24/7 running workers that also each have varying build-times. The linux machines in turn build two configurations, and one of them is a good candidate to be assisted with by EC2. But the linux builds complete much faster than the Windows one for example, so there is no problem in having a small queue for the linux job as long as it is drained in time for the Windows workers to complete their job.

**tldr;**
- not everyone needs to aim for an empty queue all the time, but rather
- sometimes it is beneficial to level out one-time setup costs by **ensuring multiple builds are done on the same instance**
- hence the request for **a way to specify minimum queue length** or some multiplicator to adjust the default trigger to complement the NoDelay method with an opposite "relaxed" fleet management

**Comment by terma** [comment]
Unrelated, but really interesting. Do you use plugin for Windows machines? If so, could you please share your Jenkins to Windows connection approach? thx

**Comment by terma** [comment]
I think you can achieve this right now with default Jenkins strategy (just disable ```No Delay Provision```) and small configuration https://support.cloudbees.com/hc/en-us/articles/115000060512-New-Shared-Agents-Clouds-are-not-being-provisioned-for-my-jobs-in-the-queue-when-I-have-agents-that-are-suspended

**Comment by terma** [comment]
To be able implement requested behavour in plugin we can make ```1``` https://github.com/jenkinsci/ec2-fleet-plugin/blob/master/src/main/java/com/amazon/jenkins/ec2fleet/NoDelayProvisionStrategy.java#L49 configurable in Fleet, so you will be able to say how many pending jobs in queue should trigger scaling.
